packages/gatsby-plugin-google-analytics/src/index.js

```diff
@@ -53,4 +53,31 @@ OutboundLink.propTypes = {
   onClick: PropTypes.func,
 }
 
-export { OutboundLink }
+/**
+ * This allows the user to create custom events within their Gatsby projects.
+ * @param {string} category Required - The object that was interacted with (e.g.video)
+ * @param {string} action Required - Type of interaction (e.g. 'play')
+ * @param {string} label Optional - Useful for categorizing events (e.g. 'Spring Campaign')
+ * @param {string} value Optional - Numeric value associated with the event. (e.g. A product ID)
+ *
+ * @link https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#events
+ */
+function CustomTrackingEvent({ category, action, label, value }) {
+  if (window.ga) {
+    window.ga(`send`, `event`, {
+      eventCategory: category,
+      eventAction: action,
+      eventLabel: label,
+      eventValue: value,
+    })
+  }
+}
+
+CustomTrackingEvent.propTypes = {
+  category: PropTypes.string.isRequired,
+  action: PropTypes.string.isRequired,
+  label: PropTypes.string,
+  value: PropTypes.number,
+}
```

Comment:
PropTypes are not enforced for non-component entities.

Reply:
What do you mean by this?

Reply:
Since your function is not a React Component the PropTypes library doesn’t do anything with it, i.e. this declaration will not be enforced.

Reply:
Ah, I was unaware of this. So I need to get it gone then?

Reply:
It’s safe to remove, yes.

Reply:
Thanks :)

Reply:
@coreyward I have added target/nonInteraction events. Google sets the target automatically unless specified. Is the best way to manage as I have, or is it OK to pass null to this function? (unsure how to approach)